### PR TITLE
New package: signing party-2.11_1

### PR DESCRIPTION
**Pull request by jcgruenhage**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### TODOs
 - [x] fix cross compilation
 - [x] <del>fix bug in perl-Interface-GnuPG (version detection is broken, which causes it to refuse operation)</del> Bug has not been found, but the check has been disabled. xbps makes sure we have a new enough version of GnuPG installed.

### CODE REVIEW

File: srcpkgs/signing-party/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+hostmakedepends="perl autoconf automake"
+makedepends="libmd-devel"
+depends="perl-Class-MethodMaker perl-GnuPG-Interface perl-MailTools perl-MIME-tools perl-Net-IDN-Encode perl-Term-ReadKey perl-Text-Template qprint"
+short_desc="OpenPGP tools for signing keys, keyring analysis, and party preparation"
+maintainer="Jan Christian Grünhage <jan.christian@gruenhage.xyz>"
+license="BSD-3-Clause, GPL-2.0-or-later, GPL-2.0-only, MIT, BSD-2-Clause, GPL-3.0-or-later"
+homepage="https://salsa.debian.org/signing-party-team/signing-party"
+distfiles="https://salsa.debian.org/${pkgname}-team/${pkgname}/-/archive/v${version}/${pkgname}-v${version}.tar.gz"
+checksum=e339cfea9af6f12bf09de8aa6c8972c168d2eb712cb72a8a7415e3f7ecb43cf2
+nocross="Super weird stacked build systems which call configure from make, while not accounting for cross compilation - see: https://github.com/void-linux/void-packages/pull/31552/checks?check_run_id=2860372307"
```
Comment (the-eater) [inline srcpkgs/signing-party/template:16]: link does not work :(

File: srcpkgs/perl-Class-MethodMaker/template [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'perl-Class-MethodMaker'
+pkgname=perl-Class-MethodMaker
+version=2.24
+revision=1
+wrksrc="${pkgname/perl-/}-${version}"
+build_style=perl-module
+hostmakedepends="perl"
+makedepends="perl"
+depends="$makedepends"
```
Comment (the-eater) [inline srcpkgs/perl-Class-MethodMaker/template:9]: consistency: use `${makedepends}`

**Comment by the-eater** [comment]
call me out if I'm wrong here, but no prefix should be needed for `Public Domain`

File: srcpkgs/signing-party/template (hunk 1 above)
Comment (jcgruenhage) [reply srcpkgs/signing-party/template:16]: arg, does github remove old builds?? :/

**Comment by jcgruenhage** [comment]
> call me out if I'm wrong here, but no prefix should be needed for `Public Domain`

We require the `custom:` prefix for everything that's not an SPDX license identifier, and while there are a few public domain alike licenses, there is not really the one, so this needs to be there

**Comment by the-eater** [comment]
on IRC I have pointed them to

```
license A string matching the license's SPDX Short identifier, Public Domain, or string prefixed with custom: for other licenses.
```

LGTM

**Comment by Chocimier** [comment]
Cross compilation works when keyanalyze is build separately:

```
post_patch() {
       vsed -i Makefile -e 's/keyanalyze//'
}
```

``` bash
# Template file for 'keyanalyze'
pkgname=keyanalyze
version=2.11
revision=1
wrksrc="signing-party-v${version}"
build_wrksrc="keyanalyze/pgpring"
build_style=gnu-configure
make_install_args="-C .."
hostmakedepends="autoconf automake"
makedepends="libmd-devel"
short_desc="Analyses the relationships between the keys in public key ring"
maintainer=" <@>"
license="GPL-2.0-only"
homepage="https://salsa.debian.org/signing-party-team/signing-party"
distfiles="https://salsa.debian.org/signing-party-team/signing-party/-/archive/v${version}/signing-party-v${version}.tar.gz"
checksum=e339cfea9af6f12bf09de8aa6c8972c168d2eb712cb72a8a7415e3f7ecb43cf2

pre_configure() {
	autoreconf -fi
}

post_build() {
	cd ..
	vsed -i Makefile -e '/all:/s:pgpring/pgpring::'
	make CFLAGS="$CFLAGS"
}
```

**Comment by jcgruenhage** [comment]
@Chocimier Thanks! I'll include that.

Aside of fixing crosscompiling, I've noticed another problem, in perl-GnuPG-Interface, it fails to get the currently installed gpg version, which means that more elaborate usage of the tools (which actually call that library) fail with an error about that. Anyone with knowledge of perl magic available to maybe take a look? I worked around this locally by overriding the version check, but that surely wouldn't be the way to properly fix this.

**Comment by jcgruenhage** [comment]
So, a few things here:
 - can we reopen this?
 - @Chocimier keyanalyze does build with that template, but the resulting package is broken. I've commented it out for now, because keyanalyze isn't really all that useful for me anyway
 - the problem with perl-GnuPG-Interface still persists, any guesses what's going wrong there?
 ```text
 error determining fileno for STDIN: Illegal seek at (eval 270) line 55.
Use of uninitialized value $line in pattern match (m//) at /usr/share/perl5/vendor_perl/GnuPG/Interface.pm line 827.
Use of uninitialized value $a in split at /usr/share/perl5/vendor_perl/GnuPG/Interface.pm line 841.
Use of uninitialized value $a in split at /usr/share/perl5/vendor_perl/GnuPG/Interface.pm line 841.
GnuPG Version 1.4 or 2.2+ required at (eval 270) line 55.
```

Here's the code:
```perl
sub _version {
    my ( $self ) = @_;

    my $out = IO::Handle->new;
    my $handles = GnuPG::Handles->new( stdout => $out );
    my $pid = $self->wrap_call( commands => [ '--no-options', '--version' ], handles => $handles );
    my $line = $out->getline;
    $line =~ /(\d+\.\d+\.\d+)/;

    my $version = $1;
    unless ($self->cmp_version($version, '2.2') >= 0 or
        ($self->cmp_version($version, '1.4') >= 0 and $self->cmp_version($version, '1.5') < 0 )) {
        croak "GnuPG Version 1.4 or 2.2+ required";
    }
    waitpid $pid, 0;

    return $version;
}
```

And here's the output of `gpg --no-options --version`:
```text
gpg (GnuPG) 2.2.34
libgcrypt 1.10.1
Copyright (C) 2022 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/jcgruenhage/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

From my understanding of perl, this looks like it should work. I've also not been able to find any patches that other distros that have this packaged apply.

**Comment by jcgruenhage** [comment]
Update on the current situation:
 - The remaining problematic package (keyanalyze) has been removed from the scope of this PR. I do not use it, and I haven't heard of anyone else doing so, so I'd be inclined to just skip it entirely.
 - The patch to perl-GnuPG-Interface was modified not to comment out lines but to remove them, to return a more up2date version (not returning a version breaks other parts, which would require a bigger patch), and to also remove the tests that started breaking because of the changes in the patch.
 - This has been tested with #38021, works as expected.
 
afaict, this addresses all the remaining feedback. I'd be happy if we could get this merged soon now :)

**Pull request merged**